File: lab1/pythonProject/core/core/FunRand.py
import random
import math

class FunRand:
    @staticmethod
    def exponential(time_mean: float) -> float:
        a = 0
        while a == 0:
            a = random.random()
        return -time_mean * math.log(a)

    @staticmethod
    def uniform(time_min: float, time_max: float) -> float:
        a = 0
        while a == 0:
            a = random.random()
        return time_min + a * (time_max - time_min)

    @staticmethod
    def normal(time_mean: float, time_deviation: float) -> float:
        return time_mean + time_deviation * random.gauss(0, 1)

    @staticmethod
    def erlang(time_mean: float, shape: float) -> float:
        a = 0
        for _ in range(int(shape)):
            a += math.log(random.random())
        return (-1 / (time_mean / shape)) * a

File: lab4/msLab4/src/FunRand.java
import java.util.Random;

public class FunRand {
    /**
     * Generates a random value according to an exponential
     * distribution
     *
     * @param delayMean mean value
     * @return a random value according to an exponential
     * distribution
     */
    public static double Exp(double delayMean) {
        double a = 0;
        while (a == 0) {
            a = Math.random();
        }
        a = -delayMean * Math.log(a);

        return a;
    }

    /**
     * Generates a random value according to a uniform
     * distribution
     *
     * @param Min min value
     * @param Max max value
     * @return a random value according to a uniform distribution
     */
    public static double Unif(double Min, double Max) {
        double a = 0;
        while (a == 0) {
            a = Math.random();
        }
        a = Min + a * (Max - Min);

        return a;
    }

    /**
     * Generates a random value according to a normal (Gauss)
     * distribution
     *
     * @param delayMean mean value
     * @param delayDev standard deviation of normal distribution
     * @return a random value according to a normal (Gauss) distribution
     */
    public static double Norm(double delayMean, double delayDev) {
        double a;
        Random r = new Random();
        a = delayMean + delayDev * r.nextGaussian();

        return a;
    }
}
